ee/server/service/orbit.go

```diff
@@ -0,0 +1,198 @@
+	"github.com/google/uuid"
+)
+
+func (svc *Service) GetOrbitSetupExperienceStatus(ctx context.Context, orbitNodeKey string, forceRelease bool) (*fleet.SetupExperienceStatusPayload, error) {
+	// this is not a user-authenticated endpoint
+	svc.authz.SkipAuthorization(ctx)
+	host, err := svc.ds.LoadHostByOrbitNodeKey(ctx, orbitNodeKey)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "loading host by orbit node key")
+	}
+
+	// get the status of the bootstrap package deployment
+	bootstrapPkg, err := svc.ds.GetHostMDMMacOSSetup(ctx, host.ID)
+	if err != nil && !fleet.IsNotFound(err) {
+		return nil, ctxerr.Wrap(ctx, err, "get bootstrap package status")
+	}
+
+	// NOTE: bootstrapPkg can be nil if there was none to install.
+	var bootstrapPkgResult *fleet.SetupExperienceBootstrapPackageResult
+	if bootstrapPkg != nil {
+		bootstrapPkgResult = &fleet.SetupExperienceBootstrapPackageResult{
+			Name:   bootstrapPkg.BootstrapPackageName,
+			Status: bootstrapPkg.BootstrapPackageStatus,
+		}
+	}
+
+	// get the status of the configuration profiles
+	cfgProfs, err := svc.ds.GetHostMDMAppleProfiles(ctx, host.UUID)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "get configuration profiles status")
+	}
+	var cfgProfResults []*fleet.SetupExperienceConfigurationProfileResult
+	for _, prof := range cfgProfs {
+		// NOTE: DDM profiles (declarations) are ignored because while a device is
+		// awaiting to be released, it cannot process a DDM session (at least
+		// that's what we noticed during testing).
+		if strings.HasPrefix(prof.ProfileUUID, fleet.MDMAppleDeclarationUUIDPrefix) {
+			continue
+		}
+
+		status := fleet.MDMDeliveryPending
+		if prof.Status != nil {
+			status = *prof.Status
+		}
+		cfgProfResults = append(cfgProfResults, &fleet.SetupExperienceConfigurationProfileResult{
+			ProfileUUID: prof.ProfileUUID,
+			Name:        prof.Name,
+			Status:      status,
+		})
+	}
+
+	// AccountConfiguration covers the (optional) command to setup SSO.
+	adminTeamFilter := fleet.TeamFilter{
+		User: &fleet.User{GlobalRole: ptr.String(fleet.RoleAdmin)},
+	}
+	acctCmds, err := svc.ds.ListMDMCommands(ctx, adminTeamFilter, &fleet.MDMCommandListOptions{
+		Filters: fleet.MDMCommandFilters{
+			HostIdentifier: host.UUID,
+			RequestType:    "AccountConfiguration",
+		},
+	})
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "list AccountConfiguration commands")
+	}
+
+	var acctCfgResult *fleet.SetupExperienceAccountConfigurationResult
+	if len(acctCmds) > 0 {
+		// there may be more than one if e.g. the worker job that sends them had to
+		// retry, but they would all be processed anyway so we can only care about
+		// the first one.
+		acctCfgResult = &fleet.SetupExperienceAccountConfigurationResult{
+			CommandUUID: acctCmds[0].CommandUUID,
+			Status:      acctCmds[0].Status,
+		}
+	}
+
+	// get status of software installs and script execution
+	res, err := svc.ds.ListSetupExperienceResultsByHostUUID(ctx, host.UUID)
```

Comment:
Double-checking my understanding of the design, the statuses reported by this will ultimately depend on the work to be done in this [subtask](https://github.com/fleetdm/fleet/issues/22381), where the script/software install results handlers will be modified to update the status of this table when the results relate to a setup experience, right?

Reply:
Correct, that's also my understanding of the flow. I believe this is what was referred to as the "callbacks" that hook into the results of software installs/scripts and update this table, so for the status endpoint we should be good with just looking up in that setup experience-specific table. @jahzielv to confirm.

Reply:
yes, that's right @gillespi314 !